clang/lib/CodeGen/CGHLSLRuntime.h

```diff
@@ -87,6 +87,7 @@ class CGHLSLRuntime {
   GENERATE_HLSL_INTRINSIC_FUNCTION(SDot, sdot)
   GENERATE_HLSL_INTRINSIC_FUNCTION(UDot, udot)
   GENERATE_HLSL_INTRINSIC_FUNCTION(WaveIsFirstLane, wave_is_first_lane)
+  GENERATE_HLSL_INTRINSIC_FUNCTION(WaveReadLaneAt, wave_readlaneat)
```

Comment:
It is probably worth naming this `wave_readlane` to align a bit with `amdgcn.readlane`. I'm not sure what the equivalent NV PTX intrinsic is... probably some crazy bundle of shuffles which confuses me.